duckduckgo/Android
# Fix issue 347 by adding plausibility check to URI schemes

Task/Issue URL: #347 

**Problem:** Anything before the first instance of a ‘:’ is interpreted as a scheme, even if it is not plausible for it to be a valid scheme.  For example, for “foo site:duckduckgo.com”, the scheme is thought to be “foo site”; for “:wq”, the scheme is thought to be “”.  Hence, both examples are erroneously treated as intent links.

**Solution:** I thought this problem was a good candidate for a regular expression.  Before trying to process the uri as an intent, the “scheme” is now checked against a regex pattern to see if its validity is plausible.  If it is not plausible, a search query is launched.  This brings the behavior in line with how I would expect the browser to handle various schemes.

**Thoughts:** It bothers me that there are two lines in the when statement that both launch a search query.  Is there possibly a better way to write this?  I am a newcomer to Kotlin and I have not investigated this much as I would like to hear some thoughts on this approach before I delve deeper into this.

**Steps to test this PR:** 
**1.** Test various invalid schemes that do not conform to the regex pattern and verify that they result in a search query.
**2.** Test intent links and other valid/invalid schemes that conform to the regex pattern and verify that they behave as expected.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
